lib/doorkeeper/oauth/token_introspection.rb

```diff
@@ -149,9 +149,9 @@ def failure_response
       #
       def active?
         if authorized_client
-          valid_token? && token_introspection_allowed?(authorized_client.application)
+          valid_token? && token_introspection_allowed?
         else
-          valid_token? && token_introspection_allowed?(authorized_token&.application)
```

Comment:
I'm not for the `authorized_token&.application` as the client send to config block. client param is should only used for `authorized_client`, not `application` of `authorized_token`.
Because at here, the request is authorized by authorized_token method, so the request authorized by authorized_token method will accidentally read the config of client_credential method (for example: token.application == authorized_client)